### PR TITLE
ocsp stapling improvements

### PR DESCRIPTION
**Pull request by rizlik**
# Description

Improvements of OCSP stapling response handling

### CODE REVIEW

**Comment by dgarske** [comment]
Retest this please

**Comment by rizlik** [comment]
Retest this please

File: wolfcrypt/src/asn.c [hunk 1]
```diff
@@ -24040,6 +24056,93 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
     return ret;
 }
 
+int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
+{
+    return ParseCertRelativeEx(cert, type, verify, cm, NULL);
```
Comment (JacobBarthelmeh) [inline wolfcrypt/src/asn.c:24061]: For local functions I prefer not to have an Ex version (reduces call stack if we can just change it).

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]

File: src/internal.c [hunk 2]
```diff
@@ -14930,6 +14940,45 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
...
+                        }
+                        ret = AllocDer(&derBuffer, cert->length, CA_TYPE, ssl->heap);
+                        if (ret != 0 || derBuffer == NULL) {
+                            FreeDecodedCert(&dCertAdd);
+                            goto exit_ppc;
+                        }
+                        XMEMCPY(derBuffer->buffer, cert->buffer, cert->length);
+                        s = MakeSigner(SSL_CM(ssl)->heap);
+                        if (s == NULL) {
+                            FreeDecodedCert(&dCertAdd);
```
[the remark this answers is not in the document]
Comment (rizlik) [reply src/internal.c:14963]: good find! Thanks

**Review by JacobBarthelmeh — APPROVED** [review]

File: src/ocsp.c [hunk 3]
```diff
@@ -584,6 +597,12 @@ static int CheckOcspResponderChain(OcspEntry* single, DecodedCert *cert,
             passed = 1;
             break;
         }
+        ca = GetCAByName(cm, ca->issuerNameHash);
+#if defined(HAVE_CERTIFICATE_STATUS_REQUEST_V2)
+    if (ca == NULL && pendingCAs != NULL) {
+        ca = findSignerByName(pendingCAs, single->issuerHash);
+    }
```
Comment (julek-wolfssl) [inline src/ocsp.c:604]: tab++

**Review by julek-wolfssl — APPROVED** [review]

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**